pkg/plugin/driver/base_test.go

```diff
@@ -101,3 +104,132 @@ func TestSkipCleanup(t *testing.T) {
 		t.Error("Expected SkipCleanup to be true but was false")
 	}
 }
+
+func envContains(env []v1.EnvVar, item v1.EnvVar) bool {
+	for _, v := range env {
+		if v == item {
+			return true
+		}
+	}
+	return false
+}
+
+func TestCreateWorkerContainerDefinition(t *testing.T) {
+	aggregatorURL := "aggregatorUrl"
+	cert := &tls.Certificate{Certificate: [][]byte{}}
+	command := []string{"sonobuoy"}
+	args := []string{"worker", "--global"}
+
+	b := &Base{
+		Definition: manifest.Manifest{
+			SonobuoyConfig: manifest.SonobuoyConfig{
+				PluginName: "test-plugin",
+				ResultType: "test-plugin",
+			},
+		},
+		SonobuoyImage:   "sonobuoy:v1",
+		ImagePullPolicy: "Never",
+		SessionID:       "sessionID",
+	}
+
+	wc := b.CreateWorkerContainerDefintion(aggregatorURL, cert, command, args)
+
+	t.Run("Fields set correctly", func(t *testing.T) {
+		if wc.Name != "sonobuoy-worker" {
+			t.Errorf("expected worker container name to be %q, but got %q", "sonobuoy-worker", wc.Name)
+		}
+
+		if wc.Image != "sonobuoy:v1" {
+			t.Errorf("expected worker container image to be %q, but got %q", "sonobuoy:v1", wc.Image)
+		}
+
+		if wc.ImagePullPolicy != v1.PullNever {
+			t.Errorf("expected worker container pull policy to be %q, but got %q", v1.PullNever, wc.ImagePullPolicy)
+		}
+
+		for i, c := range wc.Command {
+			if c != command[i] {
+				t.Errorf("expected command item %v to be %q, got %q", i, command[i], c)
+			}
+		}
+
+		for i, arg := range wc.Args {
+			if arg != args[i] {
+				t.Errorf("expected args item %v to be %q, got %q", i, args[i], arg)
+			}
+		}
+	})
+
+	t.Run("Environment constructed correctly", func(t *testing.T) {
+		expectedEnvVars := []v1.EnvVar{
+			{
+				Name:  "RESULTS_DIR",
+				Value: "/tmp/results",
+			},
+			{
+				Name:  "RESULT_TYPE",
+				Value: b.GetResultType(),
+			},
+			{
+				Name:  "MASTER_URL",
+				Value: aggregatorURL,
+			},
+			{
+				Name:  "CA_CERT",
+				Value: "",
+			},
+		}
+		for _, e := range expectedEnvVars {
+			if !envContains(wc.Env, e) {
+				t.Errorf("expected container environment to contain %q", e)
+			}
+		}
+		for _, e := range wc.Env {
+			if e.Name == "NODE_NAME" {
```

Comment:
nit: would prefer a `switch` instead of an `if...else...if else....if else...`

Reply:
Sure, and thanks for the suggestion. I'm not particularly happy with this test :-/

Reply:
It is a little odd the way you have subtests but it isn't a table driven test.

The other options it seems:
 - It is testing a single large object so you could dump it into one big test instead of bothering with the sub-tests. The error messages would just provide the context
 - make a table driven test/subtests and split off the larger chunks of logic (which make it hard to read) into little helper funcs. So you'd have:
```
tc := testCases{
  desc string
  expectFunc func(<types we're checking>) error
}{
  { desc: "Env vars set",
    expectFunc: checkEnvVars},
...
}
```

The benefit to (1) is that its straight forward to code but the downside it isn't clear when reading any particular section of the test what has been tested, will be tested, could be added, etc.

The benefit of (2) is that you can look at the table and atleast see a list of all the things you intend to test about it (even if only vague details). The downside is you have to reorg the code a bit to put those test functions in an accessible place and it may seem a bit odd since you won't be reusing those functions elsewhere. But you could, within that test itself, define those functions inline and then only make the table at the end:

```
func TestXYZ(t *testing.T){
  checkEnvVars:= func(...) error{...}
  checkFields:= func(...) error{...}

  tc := testCases {...}{...}
...
```

I think I would probably prefer (2) if I were writing it, but I wont block it either way.

Reply:
I decided to change this test and do it the way you mentioned in (2). Let me know what you think.